modules/fundamental/src/source_document/model/mod.rs

```diff
@@ -56,3 +66,34 @@ impl TryInto<StorageKey> for SourceDocument {
         (&self).try_into()
     }
 }
+
+#[async_trait::async_trait]
```

Comment:
Not sure we actually need this. I think this should be part of the actual "delete" operations, and not really exposed to the user of the (service) API.